File: algs4/dataStructure/Bag.py
class Bag(object):
    def __init__(self):
        self.content = []
        self.compass = -1

    def __iter__(self):
        return self

    def __next__(self):
        self.compass += 1
        
        if self.compass + 1 > len(self.content):
            self.compass = -1
            raise StopIteration()

        return self.content[self.compass]

    def __str__(self):
        str = ''
        for i in self:
            str + i
        return str

    def add(self, ele):
        self.content.append(ele)

    def isEmpty(self):
        return self.size == 0

    def size(self):
        return len(self.content)

    def main():
        print(Bag)

if __name__ == '__main__':
    Bag.main()

File: algs4.js/dataStructure/Bag.js
/*
 * @Author: zhenbang.tang 
 * @Date: 2018-05-29 19:45:01 
 * @Last Modified by:   zhenbang.tang 
 * @Last Modified time: 2018-05-29 19:45:01 
 * 背包
 */

class Bag {
  constructor() {
    this.content = [];
  }

  [Symbol.iterator]() {
    let index = 0;

    return {
      next:() => {
        let value = {
          value: this.content[index],
          done: index > this.content.length - 1
        }
        index ++;
        return value;
      }
    }
  }

  toString() {
    return this.content.reduce((str, item) => {
      return str.toString() + item.toString();
    })
  }

  add(ele) {
    this.content.push(ele);
  }

  get isEmpty() {
    return this.size == 0
  }

  get size() {
    return this.content.length
  }
}

module.exports = Bag;